mshadow/tensor_blob.h

```diff
@@ -430,6 +496,21 @@ class TBlob {
         dev_mask_(dev_mask),
         type_flag_(DataType<DType>::kFlag) {}
   /*!
+   * \brief constructor that construct TBlob from contiguous memory
+   * \param dptr the pointer to the memory
+   * \param shape the shape of the data
+   * \param dev_mask the device mask, can be cpu::kDevMask or gpu::kDevMask
+   * \param type_flag the type flag. Can be one of enum mshadow::dtype
+   */
```

Comment:
Check if this constructor conflicts with other constructors, if so, maybe make an explicit member function